migration-strategies/piercing/app/fragments/login/src/root.tsx

```diff
@@ -3,123 +3,95 @@ import {
   component$,
   useRef,
   useStore,
-  useStyles$,
   useStylesScoped$,
 } from "@builder.io/qwik";
 import { dispatchPiercingEvent } from "piercing-library";
 
 import styles from "./root.css?inline";
 
-export const Root = component$(() => {
-  const possibleInputErrors = {
-    noSpaces: "noSpaces",
-    noSpecialCharacters: "noSpecialCharacters",
-    maxLength: "maxLength",
-    emptyInput: "emptyInput",
-  } as const;
+const enum InputErrorMessages {
```

Comment:
I felt a simple `const enum` was cleaner than doing that type dance with records to get the strings. WDYT?

Reply:
yeah, I iterated through this code a couple of times myself 😅

Ultimately (if I remember correctly, since it was a while back 😅) I went with this solution because to me it provided the nicer type definitions, so that for example in the messages object you would see something like:
![Screenshot 2022-10-21 at 11 24 11](https://user-images.githubusercontent.com/61631103/197174831-36158210-fb56-45a9-8288-53abb5456f2f.png)

Which to me looks clearer compared to what we would get with the `const enum`:
![Screenshot 2022-10-21 at 11 23 45](https://user-images.githubusercontent.com/61631103/197174906-1f165144-7478-41b7-b529-3d769438c8e0.png)

(it would be great if the typescript language service could not convert the const enums to numbers at this stage 😅)

But I don't really have a strong strong opinion on this so if you prefer it I am totally fine with going with the `const enum` 🙂

Reply:
ah... I was pretty sure `const enum` only worked with numbers and not strings 😮 

sure, your solution looks much better! 🙂